File: Fruit.cpp
#include <iostream>
#include <vector>
using namespace std;
class Fruit {
	private:
		static string longestName;
		string name;
		string colour;
	public:
		static string getLongestName();
		Fruit(string name, string colour);
		string getName();
		void setColour(string colour);
		string describe();
};
string Fruit::longestName = "";
string Fruit::getLongestName() {
	return longestName;
}
Fruit::Fruit(string name, string colour) {
	this -> name = name;
	if (name.length() > longestName.length()) {
		longestName = name;
	}
	this -> colour = colour;
}
string Fruit::getName() {
	return this -> name;
}
void Fruit::setColour(string colour) {
	this -> colour = colour;
}
string Fruit::describe() {
	return "I am a " + colour + " fruit called " + name + ".";
}
int main()
{
	cout << "Count down from what number? ";
	int start;
	cin >> start;
	for (int i = start; i > 0; i--) {
		cout << i << ", ";
	}
	cout << "lift off!" << endl;
	vector<Fruit> fruits;
	fruits.push_back(Fruit("Apple", "Green"));
	fruits.push_back(Fruit("Strawberry", "Red"));
	fruits.push_back(Fruit("Orange", "Orange"));
	fruits.push_back(Fruit("Banana", "Yellow"));
	fruits.push_back(Fruit("Grape", "Green"));
	int fruitCount = fruits.size();
	cout << "There are " << fruitCount << " fruits..." << endl;
	for (auto fruit : fruits) {
		if (fruit.getName() == "Grape") {
			fruit.setColour("Purple");
		}
		cout << fruit.describe() << endl;
	}
	cout << "The longest fruit name is " << Fruit::getLongestName() << "!" << endl;
	return 0;
}

File: Fruit.lua
Fruit = {
	longestName = "",
	getLongestName = function()
		return Fruit.longestName
	end,
	new = function(name, colour)
		local self = {}
		self.name = name
		if string.len(name) > string.len(Fruit.longestName) then
			Fruit.longestName = name
		end
		self.colour = colour
		self.getName = function()
			return self.name
		end
		self.setColour = function(colour)
			self.colour = colour
		end
		self.describe = function()
			return "I am a " .. self.colour .. " fruit called " .. self.name .. "."
		end
		return self
	end,
}
io.write("Count down from what number? ")
start = io.stdin:read()
for i=tonumber(start),1,-1 do
	io.write(i .. ", " )
end
print ("lift off!")
fruits = { Fruit.new("Apple", "Green"),
			 Fruit.new("Strawberry", "Red"),
			 Fruit.new("Orange", "Orange"),
			 Fruit.new("Banana", "Yellow"),
			 Fruit.new("Grape", "Green") }
fruitCount = #fruits
print("There are " .. fruitCount .. " fruits...")
for _, fruit in ipairs(fruits) do
	if fruit.getName() == "Grape" then
		fruit.setColour("Purple")
	end
	print (fruit.describe())
end
print("The longest fruit name is " .. Fruit.getLongestName() .. "!")
